### PR TITLE
(GH-2219) Add 'module' and 'modulepath' topic guides

### PR DESCRIPTION
**Pull request by beechtom**
This adds a new 'module' topic guide. It gives a brief description of
what a module is and how Bolt can manage a project's modules.

This also adds a new 'modulepath' topic guide that describes what the
modulepath is, precedence of modules on the modulepath, and how Bolt
adds directories to the modulepath.

!no-release-note

### CODE REVIEW

File: guides/module.txt
Comment (lucywyman) [inline guides/module.txt:11]: It might be useful to just have a separate modulepath guide. As a new person, it seems like it'd be confusing to think about both the `modules` configured in the project and the modulepath. Something like "to learn more about how modules are loaded, run 'bolt guide modulepath'" and then getting more in depth about the default path and what special paths are (i.e. `modules` is for custom modules, `.modules` is where modules get installed, etc.).

Comment (hestonhoffman) [reply guides/module.txt:11]: Oh that's a great idea!

**Review by lucywyman — CHANGES_REQUESTED** [review]
Breakout modulepath into separate guide.

**Comment by beechtom** [comment]
Added a modulepath guide. Left out listing specific directories on the default modulepath since that gets more into documentation and is also made a little more confusing because we currently have two default modulepaths.

File: guides/modulepath.txt
Comment (hestonhoffman) [inline guides/modulepath.txt:12]:
```suggestion
    or configured modulepath, Bolt automatically adds two directories to
```
It's just the two now right? If it's a small, but undefined number, I would maybe use different phrasing - For example "certain directories" / "specific directories" or forego the descriptor and just use "...adds directories...".

Comment (beechtom) [reply guides/modulepath.txt:12]: _ish_. It adds `.modules` if you're using the module management workflow, and the project if you're using a named project. I'll be less specific.

File: guides/modulepath.txt
Comment (hestonhoffman) [inline guides/modulepath.txt:12]:
```suggestion
    or configured modulepath, Bolt automatically adds directories to the
```

File: guides/modulepath.txt
Comment (lucywyman) [inline guides/modulepath.txt:20]:
```suggestion
    and instead will ignore modules with a lower precedence.
```

**Review by lucywyman — COMMENTED** [review]
One nit, otherwise LGTM!

**Review by lucywyman — APPROVED** [review]

**Comment by lucywyman** [comment]
Will merge once Heston approves!

File: guides/modulepath.txt
Comment (hestonhoffman) [reply guides/modulepath.txt]: Ah, good catch!

**Review by hestonhoffman — APPROVED** [review]

**Pull request merged**
